jsonParser.go

```diff
@@ -130,39 +128,135 @@ type Link struct {
 	ContentVersion      string `json:"content-version"`
 	IntendedApplication string `json:"intended-application"`
 }
-
 type Primary struct {
 	URL string `json:"URL"`
 }
 type Resource struct {
 	Primary Primary `json:"primary"`
 }
-type Issued struct {
-	DateParts [][]int `json:"date-parts"`
-}
 type JournalIssue struct {
-	Issue           *string          `json:"issue"`
-	PublishedOnline *PublishedOnline `json:"published-online"`
-	PublishedPrint  *PublishedPrint  `json:"published-print"`
+	Issue           *string    `json:"issue"`
+	PublishedOnline *DateParts `json:"published-online"`
+	PublishedPrint  *DateParts `json:"published-print"`
 }
 type IssnType struct {
 	Value string `json:"value"`
 	Type  string `json:"type"`
 }
-type PublishedOther struct {
-	DateParts [][]int `json:"date-parts"`
-}
-type Published struct {
-	DateParts [][]int `json:"date-parts"`
-}
-
 type Relation struct {
 	Cites []any `json:"cites"`
 }
-
 type License struct {
 	URL            string  `json:"URL"`
 	Start          Indexed `json:"start"`
 	DelayInDays    int     `json:"delay-in-days"`
 	ContentVersion string  `json:"content-version"`
 }
+
+//
+func (pub *CrossRef) GetSimpleTitle() []string {
+	simpleTitle := pub.Title
+	if len(simpleTitle) == 0 {
+		return []string{""}
+	}
+	for pos, title := range simpleTitle {
+		title = strings.Replace(title, "\n", " ", -1)
+		title = strings.Replace(title, "( )+", " ", -1)
+		simpleTitle[pos] = strings.TrimSpace(title)
+	}
+	return simpleTitle
+}
+
+func (pub *CrossRef) GetSimpleDOI() string {
+	return pub.Doi
+}
+
+func (pub *CrossRef) GetSimpleAuthor() string {
+	if len(pub.Author) == 0 {
+		return ""
+	}
+	var simpleAuthor strings.Builder
+	for _, auth := range pub.Author {
+		simpleAuthor.WriteString(*auth.Family + " ")
+	}
+	return strings.TrimSpace(simpleAuthor.String())
+}
+
+func (pub *CrossRef) GetSimpleFirstAuthor() string {
+	if len(pub.Author) == 0 {
+		return ""
+	}
+	for _, auth := range pub.Author {
+		if *auth.Sequence == "first" && *auth.Family != "" {
+			return *auth.Family
+		}
+	}
+	// not sequence information apparently, so as fallback we use the first
+	// author in the author list
+	return *pub.Author[0].Family
+}
+
+func (pub *CrossRef) GetSimpleFirstPage() string {
+	sp := regexp.MustCompile(`,|-|\s`)
```

Comment:
The data is not this simple. Page can be both a single number and a range. And I think you should use `strings.Split()` if possible to avoid using regex

Reply:
I don't know how the data looks like in general but this is a translation of the .js function.